doc: update config (#15)
seri037/llcppg: llpkgstore.md

### Field description

**upstream**

| key | type | defaultValue | optional | description |
|------|------|--------|------|------|
| installer | `string` | "conan" | ✅ | upstream provide platform |
| config.options | `string` | "" | ✅ | platform CLI config |
| package.name | `string` | - | ❌ | package name in platform |

> | package.version | `string` | - | ✅ | package original version |

Reviewer: ```suggestion
| key | type | defaultValue | optional | description |
|------|------|--------|------|------|
| installer | `string` | "conan" | ✅ | upstream binary provider |
| config | `map[string]string` | [] | ✅ | config of installer |
| package.name | `string` | - | ❌ | package name in platform |
| package.version | `string` | - | ❌ | original package version |
```

**generator** 

> | 字段 | 类型 | 默认值 | 必填 | 说明 |

Reviewer 2: Unnecessary Chinese in the document.
```
| key         | type    | default value   | optional | description           |
```

Reviewer 2: It seems that it has been fixed

> | 字段 | 类型 | 默认值 | 必填 | 说明 |

Reviewer 3: It should be in English, not in Chinese.

|------|------|--------|------|------|
| name | `string` | "llcppg" | ✅ | generator name |
| version | `string` | "latest" | ✅ | generator version |
#### For developers

If no `upstream.package.version` is specified, the `conan search` command will be used to fetch all available versions of the current C package. You can then manually select the version from the command line.

> 

Reviewer: It shouldn't be optional.
```suggestion
```

**Currently**, the cfg system supports third-party libraries for C/C++ **only**. Support for other languages, such as Python and Rust, may be added in the future, but there are no updates at this time. 

For C/C++, the only supported generator is llcppg. IMPORTANT: llcppg is still in development and does not have strict versioning, so it is recommended to use `latest` as the default value for configuration.

> At the moment, we heavily rely on Conan as the upstream distribution platform for C libraries. Therefore, Conan is the only installer supported for C libraries. We have plans to introduce more distribution platforms in the future to provide broader coverage.

Reviewer: ```suggestion
At the moment, we heavily rely on Conan as the upstream distribution platform for C libraries. Therefore, Conan is the only installer supported for C libraries. This field exists for better extensibility and a possible situation that Conan's service might be unavailable in the future. We have planned to introduce more distribution platforms in the future to provide broader coverage.
```

### Field description

**upstream**

| key | type | defaultValue | optional | description |
|------|------|--------|------|------|
| installer | `string` | "conan" | ✅ | upstream binary provider |
| config | `map[string]string` | [] | ✅ | config of installer |
| package.name | `string` | - | ❌ | package name in platform |
| package.version | `string` | - | ❌ | original package version |

**generator** 

| key | type | defaultValue | optional | description |
|------|------|--------|------|------|
| name | `string` | "llcppg" | ✅ | generator name |
| version | `string` | "latest" | ✅ | generator version |
#### For developers

> When you specify a version of a C library, we will check the corresponding upstream platform to see if the version is supported. If the version is not found, the CLI will provide you with an opportunity to reselect a version supported by the current upstream platform.
> 

Reviewer: We have no plan to make it work in this way. Shouldn't it be discussed first?
```suggestion
```

> **IMPORTANT**: For automated workflows, if an incorrect version is chosen and there is no user intervention (i.e., no input in the CLI's stdin), an error will be automatically thrown and the process will terminate after a timeout (usually 5 seconds).
> 

Reviewer: Not expected. Should terminate immediately when error thrown.
```suggestion
```

At the moment, we heavily rely on Conan as the upstream distribution platform for C libraries. Therefore, Conan is the only installer supported for C libraries. This field exists for better extensibility and a possible situation that Conan's service might be unavailable in the future. We have planned to introduce more distribution platforms in the future to provide broader coverage.  

> During the development and debugging phase, any changes made to the original configuration file will be displayed in the final output (stdout). Before proceeding, the CLI will prompt you to confirm whether you want to save the modified version locally.

Reviewer: Not expected. It impossible to debug manually in a Github Action. All CLI operation should be done in another part (e.g. package cmd). If you're thinking of debugging locally, it's a bit off topic.
```suggestion
```
